When I `export DOCKER_CROSSPLATFORMS="linux/386 linux/arm darwin/amd64"`, I got these errors during `make cross` 

github.com/docker/docker/pkg/devicemapper
pkg/devicemapper/devmapper.go:85: undefined: cdmTask
# github.com/docker/docker/pkg/loopback
pkg/loopback/attach_loopback.go:21: undefined: LoNameSize
pkg/loopback/attach_loopback.go:21: invalid array bound LoNameSize
pkg/loopback/attach_loopback.go:22: undefined: LoNameSize
pkg/loopback/attach_loopback.go:22: invalid array bound LoNameSize
pkg/loopback/ioctl.go:25: undefined: loopInfo64
pkg/loopback/ioctl.go:39: undefined: loopInfo64
Makefile:114: recipe for target 'cross' failed
make: *** [cross] Error 1
aragunathan@aragunathan-XPS-13-9343:~/code/src/github.com/docker/docker$ echo $DOCKER_CROSSPLATFORMS
linux/386 linux/arm darwin/amd64

--- a/pkg/devicemapper/devmapper.go
+++ b/pkg/devicemapper/devmapper.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,cgo
 
 package devicemapper
 

--- a/pkg/devicemapper/devmapper_log.go
+++ b/pkg/devicemapper/devmapper_log.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,cgo
 
 package devicemapper
 

--- a/pkg/devicemapper/devmapper_wrapper.go
+++ b/pkg/devicemapper/devmapper_wrapper.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,cgo
 
 package devicemapper
 

--- a/pkg/devicemapper/devmapper_wrapper_deferred_remove.go
+++ b/pkg/devicemapper/devmapper_wrapper_deferred_remove.go
@@ -1,4 +1,4 @@
-// +build linux,!libdm_no_deferred_remove
+// +build linux,cgo,!libdm_no_deferred_remove
 
 package devicemapper
 

--- a/pkg/devicemapper/devmapper_wrapper_no_deferred_remove.go
+++ b/pkg/devicemapper/devmapper_wrapper_no_deferred_remove.go
@@ -1,4 +1,4 @@
-// +build linux,libdm_no_deferred_remove
+// +build linux,cgo,libdm_no_deferred_remove
 
 package devicemapper
 

--- a/pkg/devicemapper/ioctl.go
+++ b/pkg/devicemapper/ioctl.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,cgo
 
 package devicemapper
 

--- a/pkg/loopback/attach_loopback.go
+++ b/pkg/loopback/attach_loopback.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,cgo
 
 package loopback
 

--- a/pkg/loopback/ioctl.go
+++ b/pkg/loopback/ioctl.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,cgo
 
 package loopback
 

--- a/pkg/loopback/loop_wrapper.go
+++ b/pkg/loopback/loop_wrapper.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,cgo
 
 package loopback
 

--- a/pkg/loopback/loopback.go
+++ b/pkg/loopback/loopback.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,cgo
 
 package loopback
 

